Realm/Tests/ObjectTests.m

```diff
@@ -607,25 +607,59 @@ static void addProperty(Class cls, const char *name, const char *type, size_t si
 }
 
 - (void)testObjectSubclassAddedAtRuntime {
-    Class objectClass = objc_allocateClassPair(RLMObject.class, "RuntimeGeneratedObject", 0);
-    addProperty(objectClass, "objectCol", "@\"RuntimeGeneratedObject\"", sizeof(id), alignof(id), ^(__unused id obj) { return nil; });
-    addProperty(objectClass, "intCol", "i", sizeof(int), alignof(int), ^int(__unused id obj) { return 0; });
-    objc_registerClassPair(objectClass);
-    XCTAssertEqualObjects([objectClass className], @"RuntimeGeneratedObject");
+    @autoreleasepool {
+        Class objectClass = objc_allocateClassPair(RLMObject.class, "RuntimeGeneratedObject", 0);
+        addProperty(objectClass, "objectCol", "@\"RuntimeGeneratedObject\"", sizeof(id), alignof(id), ^(__unused id obj) { return nil; });
+        addProperty(objectClass, "intCol", "i", sizeof(int), alignof(int), ^int(__unused id obj) { return 0; });
+        objc_registerClassPair(objectClass);
+        XCTAssertEqualObjects([objectClass className], @"RuntimeGeneratedObject");
 
-    RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
-    configuration.objectClasses = @[objectClass];
-    XCTAssertEqualObjects([objectClass className], @"RuntimeGeneratedObject");
+        RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
+        configuration.objectClasses = @[objectClass];
+        XCTAssertEqualObjects([objectClass className], @"RuntimeGeneratedObject");
 
-    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
-    [realm beginWriteTransaction];
-    id object = [objectClass createInRealm:realm withValue:@{@"objectCol": [[objectClass alloc] init], @"intCol": @17}];
-    RLMObjectSchema *schema = [object objectSchema];
-    XCTAssertNotNil(schema[@"objectCol"]);
-    XCTAssertNotNil(schema[@"intCol"]);
-    XCTAssert([[object objectCol] isKindOfClass:objectClass]);
-    XCTAssertEqual([object intCol], 17);
-    [realm commitWriteTransaction];
+        RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
+        [realm beginWriteTransaction];
+        id object = [objectClass createInRealm:realm withValue:@{@"objectCol": [[objectClass alloc] init], @"intCol": @17}];
+        RLMObjectSchema *schema = [object objectSchema];
+        XCTAssertNotNil(schema[@"objectCol"]);
+        XCTAssertNotNil(schema[@"intCol"]);
+        XCTAssert([[object objectCol] isKindOfClass:objectClass]);
+        XCTAssertEqual([object intCol], 17);
+        [realm commitWriteTransaction];
+    }
+    // Creating the class via ObjC runtime can cause issues with other tests. Specifically
```

Comment:
This all seems very dubious? There shouldn't be any difference between classes created explicitly via the obj-c runtime functions and ones created via loading a bundle. If there is a difference, then this test appears to just be showing that our code is broken, and deleting the classes is just working around the bug that the test uncovered, since we do intend to support dynamically created RLMObject subclasses.

Reply:
I located the issue to be at ```
util::Optional<RLMClassInfo&> RLMSchemaInfo::operator[](realm::TableKey const& key) {
    for (auto& pair : m_objects) {
        if (pair.second.objectSchema->table_key == key)
            return pair.second;
    }
    return util::none;
}
```
where the `table_key` is a garbage value for this runtime registered object. So I suppose guarding against that somehow is the fix